### PR TITLE
mods to support blacklisted android devices

### PR DESCRIPTION
**Pull request by Cy-Bo-Rg**
I've added again a few lines to support blacklisted android devices with black screen problem (appearing since rev68).
I've added a modified index.html called index_mobile.html which fixes
resize problems and refresh fix problems in black listed android devices on stock browser.
I've removed a variable from a demo which was declared but was not used. issue #6766

My device is a Galaxy Tab 2.0 GT-P3110 running Android 4.1.2 on a Power VR SGX-540 (blacklisted)
Almost all demos works perfectly with this mod. Without the lines added in the "src/Three.js" , the device shows a black screen.
With the lines , it shows the demos but with a wrong refresh rate (1 to 5 fps) and the windows doesn't properly resize.
With the modified index_mobile.html the window resizes correctly and the refresh rate is at its maximum possible (60+fps on some demos).
On the same device the Crosswalk Chromium webview crashes sometimes and has a lower FPS than with this fix; the same happens on Chrome (latest dev build) and Firefox (latest beta build).
To make a comparison, the "material/skin" demo runs at 2 fps on crosswalk while on android stock browser with this fix runs at 8 fps (6-5 while rotating the view).

Related issues  #2119 (unreasonably closed)
There are many related issues but I couldn't find all of them because they reside in other revisions now.

Here are some screens of the android stock browser running three.js rev71.
Without lines added in src/Three.js , current index.html.
screenshot_2015-07-04-05-57-03

With lines added in the build but without modified index. Doesn't properly resize , the frame rate is wrong and the image refreshes at 1 to 5 fps.
screenshot_2015-07-04-05-58-39

With lines added in the build and with index_mobile.html : 100% playable.
screenshot_2015-07-04-06-03-48

This video shows the problem and the fix in action.
https://www.youtube.com/watch?v=P_i0uOPWBuI
These modifications won't alter the behavior on desktop,which remains the same.

I've highly tested this refresh workaround since rev66. It always works.
But I know for sure that it is present starting from rev54 because I can run helloenjoy samsung racer s with the refresh fix.

### CODE REVIEW

**Comment by Cy-Bo-Rg** [comment]
This is a screenshot of rev72 dev with my modified index_mobile.html.
Resizes correctly and the refresh rate is good,the shadows are rendered correctly even though no controls are showing up.
screenshot_2015-07-04-17-00-50

**Comment by Cy-Bo-Rg** [comment]
The refresh problem is a webgl bug and I'm letting them know of it but the black screen and the resize problem are some three.js bugs which can be fixed by adding those lines of code which were present in previous versions and by modifying the index.html accordingly.

**Comment by Cy-Bo-Rg** [comment]
A few words are much appreciated.

**Comment by mrdoob** [comment]
Android's stock browser doesn't support requestAnimationFrame?

**Comment by Cy-Bo-Rg** [comment]
Someone says so, but he's wrong.
It's supported,at least partially.
https://youtu.be/UIualrl8FNo

**Comment by Cy-Bo-Rg** [comment]
This is three.js rev72 dev on Android 4.2.2 on Mali 400 with its stock browser.

Rev72dev without code and refresh fix
screenshot_2015-07-05-10-46-19

Rev72dev with code added and fix
screenshot_2015-07-05-11-32-30
Rev72dev with code added and fix
screenshot_2015-07-05-10-48-16
Rev72dev with code added and fix
screenshot_2015-07-05-11-34-03
Rev72dev with code added and fix
screenshot_2015-07-05-11-35-16


For a comparison with my blacklisted device I've tested the latest chrome dev on this Mali 400 (which isn't blacklisted) versus stock android browser on my Power VR SGX540.

Mali400 running "parallaxmap" demo on rev72dev
screenshot_2015-07-05-11-57-21
Mali400 running "parallaxmap" demo on rev72dev
screenshot_2015-07-05-11-57-50

Power VR SGX540 running "parallaxmap" on rev72dev (modified)
screenshot_2015-07-05-12-10-50

**Comment by Cy-Bo-Rg** [comment]
The last word is up to you guys.

**Comment by Cy-Bo-Rg** [comment]
The devices which may benefit from such enhancements are 1497.
(http://www.gsmarena.com/results.php3?sOSes=2&sOSversions=2410

Jelly bean users (which can't upgrade or who don't care about changing browser to surf the web) are the vast majority.
!(http://blogs-images.forbes.com/ewanspence/files/2015/01/lollipop-dead-graph.jpg)

**Comment by Cy-Bo-Rg** [comment]
"Android's stock browser doesn't support requestAnimationFrame?"
Xperia MT27i Pepper running andorid 2.3.7 on Mali 400 ,stock browser 2.3.7.
I'm sorry but this phone cannot even take screenshots without root access.

Current index.html with rev72dev without added code doesn't even show up (if I'm not wrong,an html tag isn't closed somewhere...)

My corrected index_mobile.html , instead, does show up correctly and as expected.

Rev72dev without mod : **black screen**(on all demos,this was a canvas demo).

Rev72dev modded.

Rev72dev modded.

Rev72dev modded.

Rev72dev modded.

Rev72dev modded.

Rev72dev modded.


What else can I say ?
I don't know if you need more evidence than this, but avoiding the conversation while talking with others...Just close the pull request , I'll understand...

**Comment by WestLangley** [comment]
> I don't know if you need more evidence than this, but avoiding the conversation while talking with others...Just close the pull request , I'll understand...

@Cy-Bo-Rg Thank you for your contributions.

Please relax and be patient. @mrdoob will get around to this in due time. Remember, it will be on his schedule, not yours.

**Comment by Cy-Bo-Rg** [comment]
Absolutely, I recognize the efforts , I'm in love with his work , that's probably why I care so much, but you need one minute out of 2 days to make a couple of comments :P

**Comment by WestLangley** [comment]
> but you need one minute out of 2 days to make a couple of comments

That is not how life works, dude. : - )

**Comment by Cy-Bo-Rg** [comment]
I get it... :P
But I apologize for my index_mobile.html though; it has the demo list of three.js master revision 71,not 72dev.

**Comment by Cy-Bo-Rg** [comment]
Sorry if I add many comments, but to have a laugh together, watch this... X)

http://caniuse.com/#feat=webgl

I mean , WHY ??? X)

**Comment by Cy-Bo-Rg** [comment]
please notice that the page "http://twofuckingdevelopers.com/2013/06/gsap-requestanimationframe-performance-on-android-browsers/" isn't available anymore and I really don't know why or what happened since two days ago it was visible and ready to be seen...
BUT, there is a snapshot in the Archive.org available to be seen...
http://web.archive.org/web/20150316151742/http://twofuckingdevelopers.com/2013/06/gsap-requestanimationframe-performance-on-android-browsers/

**Comment by mrdoob** [comment]
Would it be possible to implement the fixes of `index_mobile.html` into `index.html`? I would prefer to just have one file. Adding the requestAnimationFrame shim in `Three.js` should be ok.

**Comment by Cy-Bo-Rg** [comment]
Sure, I'm on it. I'll fix the indentation too and I'll update it with rev72 demos.
I might specify in a comment that the "refresh fix" has to be applied in the html of a demo for it to work "standalone" without the index because if you open a demo without it, it will not refresh properly unless the hidden marquee is present.
But I can even apply it to all the demos if requested to do so.
Waiting for updates.
P.s. : do I have to close this pull request and open another one,right ?

**Comment by Cy-Bo-Rg** [comment]
updated now. sorry for the late.

**Comment by Cy-Bo-Rg** [comment]
p.s: Firefox beta 40.0 has problem rendering most of the demos and with Chrome, no matter how many flags I activate, I just can't use it.

**Comment by Cy-Bo-Rg** [comment]
sorry , I've pressed the button for error :P

File: examples/index.html [hunk 1]
```diff
@@ -564,7 +589,7 @@
 
 		}
 
-		</script>
+		</script><div class="marquee"><p>refresh_fix_on_some_blacklisted_android_chipsets</p></div></article>
```
Comment (mrdoob) [inline examples/index.html]: Hold on a second. Blacklisted... does it mean that you've disabled the blacklisting? If so, wouldn't it be better to just support Firefox? I suspect very few people disable blacklisting.

Comment (Cy-Bo-Rg) [inline examples/index.html]: I've not disabled blacklisting on stock browser, it renders whatever you throw at it.
Stock browser doesn't have blacklist.
That's the point. Hundreds of millions of users would benefit.

**Comment by Cy-Bo-Rg** [comment]
The index has been modified in that way because some css properties like "calc(100%-310)" are unsupported on many browsers on mobile while "width="100%" in the tag is perfectly recognized.
This doesn't affect rendering on desktop browsers though , which are relying on the css and which are working perfectly.

**Comment by mrdoob** [comment]
Ok, adding `requestAnimationFrame` sound good, but the css changes seem kind of hacky.

**Comment by Cy-Bo-Rg** [comment]
@mrdoob , the css isn't a problem for me, I mean, without it the frame doesn't scale properly and I see this 
screenshot_2015-07-04-05-58-39
but it's not a problem considering that it can be fixed easily.
Even the refresh fix can be left alone if you don't want to add it (even though I really think of it as a useful implementation to keep backward compatibility with old chipsets).
If you could tell me what to eliminate and what to keep I'll fix the pull request a.s.a.p. today.

**Comment by Cy-Bo-Rg** [comment]
@mrdoob 
Pull request updated.
"massive hack" removed...
It was really good by the way. If for whatever reason webgl should stop refreshing,it could have prevented it from happening.
But the requestAnimationFrame part is what's really needed, so, I can live without it.

**Comment by mrdoob** [comment]
Thanks!

**Comment by Cy-Bo-Rg** [comment]
Thank you, without you this couldn't be possible :)

**Pull request merged**